### DISCUSSION TITLE
Would like to add a socket similar to the ZMQ_STREAM type in ZMQ.

### DISCUSSION THREAD

**Discussion by nanji886** [nanomsg/nng#1860] [Ideas]
Adding a ZMQ_STREAM socket makes it possible to call TCP sockets in a more uniform and tidy way.

**Comment by gdamore** [comment]
Please see the nng_stream API.  Start with [nng_stream](https://nng.nanomsg.org/man/v1.2.2/nng_stream.5.html)  You could also look at nng_stream_dialer, nng_stream_listener, and such.  I think this satisfies your request.
